nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/sql/model/Literals.java

```diff
@@ -88,9 +89,14 @@ public static String valueOf(StringLiteral lit) {
     public static Literal toLiteral(Object value) {
         if (value instanceof Boolean) {
             return new BooleanLiteral(((Boolean) value).booleanValue());
-        } else if (value instanceof Calendar || value instanceof Date || value instanceof Temporal
-                || value instanceof DateTime) {
-            return new DateLiteral(new DateTime(value));
+        } else if (value instanceof Calendar) {
+            return new DateLiteral(toZonedDateTime((Calendar) value));
+        } else if (value instanceof Date) {
+            return new DateLiteral(toZonedDateTime((Date) value));
+        } else if (value instanceof Temporal) {
+            return new DateLiteral(ZonedDateTime.from((Temporal) value));
+        } else if (value instanceof ZonedDateTime) {
```

Comment:
This condition is never reached, `ZonedDateTime` implements `Temporal`

Reply:
you're right, fixed